client-runtime/src/main/java/com/microsoft/rest/v2/http/BufferedHttpResponse.java

```diff
@@ -49,43 +47,40 @@ public HttpHeaders headers() {
     public Single<byte[]> bodyAsByteArray() {
         if (body == null) {
             body = innerHttpResponse.bodyAsByteArray()
-                    .map(new Function<byte[], byte[]>() {
-                        @Override
-                        public byte[] apply(byte[] bytes) {
-                            body = Single.just(bytes);
-                            return bytes;
-                        }
+                    .map(bytes -> {
+                        body = Single.just(bytes);
```

Comment:
I know you're not really changing this code, but just out of curiosity, isn't this a race condition? If the map is being run by a different thread, then it's possible that body will get assigned on line 49 after it gets assigned on line 51?